Sources/TuistGenerator/Generator/SchemesGenerator.swift

```diff
@@ -339,82 +366,182 @@ final class SchemesGenerator: SchemesGenerating {
+            guard let pbxTarget = generatedProject.targets[target.name] else { return nil }
+            
+            return targetBuildableReference(target: target,
+                                            pbxTarget: pbxTarget,
+                                            projectPath: generatedProject.name)
         }
 
-        var schemeActions: [XCScheme.ExecutionAction] = []
-        actions.forEach { action in
-            let schemeAction = XCScheme.ExecutionAction(scriptText: action.scriptText,
-                                                        title: action.title,
-                                                        environmentBuildable: nil)
+        let schemeAction = XCScheme.ExecutionAction(scriptText: action.scriptText,
+                                                    title: action.title,
+                                                    environmentBuildable: nil)
 
-            schemeAction.environmentBuildable = schemeBuildableReference(targetName: action.target,
-                                                                         project: project,
-                                                                         generatedProject: generatedProject)
-            schemeActions.append(schemeAction)
+        schemeAction.environmentBuildable = schemeBuildableReference(target: target,
+                                                                     generatedProject: generatedProject)
+        return schemeAction
+    }
+    
+    /// Generates the array of BuildableReference for targets that the
+    /// coverage report should be generated for them.
+    ///
+    /// - Parameters:
+    ///   - target: test actions.
+    ///   - graph: tuist graph.
+    ///   - generatedProjects: Generated Xcode projects.
+    ///   - rootPath: Root path to workspace or project.
+    /// - Returns: Array of buildable references.
+    private func testCoverageTargetReferences(target: TargetReference,
+                                              graph: Graphing,
+                                              generatedProjects: [AbsolutePath: GeneratedProject],
+                                              rootPath: AbsolutePath) throws -> XCScheme.BuildableReference? {
+        guard let (targetNode, generatedProject) = try lookupTarget(reference: target,
+                                                                graph: graph,
+                                                                generatedProjects: generatedProjects,
+                                                                rootPath: rootPath) else {
+            return nil
         }
-        return schemeActions
+        
+        guard let pbxTarget = generatedProject.targets[targetNode.target.name] else { return nil }
+        
+        return targetBuildableReference(target: targetNode.target,
+                                        pbxTarget: pbxTarget,
+                                        projectPath: generatedProject.name)
```

Comment:
`generatedProject.name`? I believe this needs to be the relative path the the `.xcodeproj` from he rootPath (having the helper method will ensure we have the correct implementation too!)